Classes/GarbageCollector.php

```diff
@@ -201,12 +201,15 @@ protected function deleteIndexDocuments($table, $uid)
         $indexQueueItems = $this->getIndexQueue()->getItems($table, $uid);
         foreach ($indexQueueItems as $indexQueueItem) {
             $site = $indexQueueItem->getSite();
+            $solrConfiguration = $site->getSolrConfiguration();
 
             // a site can have multiple connections (cores / languages)
             $solrConnections = $connectionManager->getConnectionsBySite($site);
             foreach ($solrConnections as $solr) {
                 $solr->deleteByQuery('type:' . $table . ' AND uid:' . intval($uid));
-                $solr->commit(false, false, false);
+                if ($solrConfiguration->getEnableCommits()) {
```

Comment:
I think, retrieving the setting could also be done outside the loop. What do you think?